crates/transaction-pool/src/validate/eth.rs

```diff
@@ -377,18 +383,15 @@ where
         }
 
         // If we don't have a state provider yet, fetch the latest state
-        if maybe_state.is_none() {
-            match self.client.latest() {
-                Ok(new_state) => {
-                    *maybe_state = Some(new_state);
-                }
+        let state = match maybe_state {
+            Some(state) => Box::new(state),
```

Comment:
This will now box layer every single provider,

for example batch validating 10 txs, the last provider could be `Box<Box<....Box<dyn StateProvider>`

Reply:
compiler optimises that away. that's why the trait `StateProvider` is still implemented for the type, `StateProvider` hasn't been `auto_impl` for `Box<Box<Box<Box<T>>>>` right?
see 
1) https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=c7bd2ec215bbcceb3882f72be537d074
2) https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=e49bfef8c38dd7a9b0a1909660b4131b
3) https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=db352470586b4541c65e51460d6413a4